### ISSUE TITLE
[apex]  CRUD permission issue. When done in one method in the class, other methods can skip it and does not fail the rule

### ISSUE THREAD

**Issue by genoud** [pmd/pmd#1127]
**Affected PMD versions**
6.2.0+
7.0.0 

**ApexCRUDViolation**

**When done in one method in the class, other methods can skip it and does not fail the rule**

```apex
public with sharing class AccountManager {
        public static String getAccountById(ID accountId){
            if(!Account.getSObjectType().getDescribe().isAccessible() || 
                    !Schema.sObjectType.Account.fields.Name.isAccessible()){
                    return 'You do not have access to the account object or to the name field';
             }
             Account acc=[Select Name From Account WHERE Id=:accountId];
              return acc;
        }
        public static List<Account> getAccountsByName(String name){
              String nameParam='%'+name+'%';
              //This should fail validation
              List<Account> accountList=[Select Name From Account WHere Name like :nameParam];
              return accountList;
        }
    
}
```

**Running through:** Apex PMD extension for VS Code

**Comment by jsotuyod** [comment]
This is a duplicate of #4589. Even if this report is older, the example and description on that ticket is clearer, so we are keeping that one instead.
